CMakeLists.txt

```diff
@@ -86,25 +86,20 @@ else ()
         if (USE_OPENSSL AND NOT ANDROID)
             find_package(OpenSSL REQUIRED)
             find_package(Threads REQUIRED)
-            add_library(crypto UNKNOWN IMPORTED)
-            set_target_properties(crypto PROPERTIES
-                INTERFACE_INCLUDE_DIRECTORIES "${OPENSSL_INCLUDE_DIR}")
-            set_target_properties(crypto PROPERTIES
-                IMPORTED_LINK_INTERFACE_LANGUAGES "C"
-                IMPORTED_LOCATION "${OPENSSL_CRYPTO_LIBRARY}")
-            add_dependencies(crypto Threads::Threads)
+            set(PLATFORM_LIBS OpenSSL::Crypto Threads::Threads)
             message(STATUS "Using libcrypto from system: ${OPENSSL_CRYPTO_LIBRARY}")
         elseif(NOT USE_OPENSSL AND IN_SOURCE_BUILD)
             if (TARGET crypto)
                 message(STATUS "Using libcrypto from AWS-LC")
             else()
                 message(FATAL_ERROR "Target crypto is not defined, failed to find libcrypto.")
             endif()
+            set(PLATFORM_LIBS crypto dl)
```

Comment:
why does this branch require `dl` but the `if (USE_OPENSSL AND NOT ANDROID)` branch doesn't?

I realize the code was like this when you got here, but this series of if-elses is really confusing to me. I see 3 libraries being required-or-not: crypto, thread, dl

And there are 3 toggles that lead to them being selected: USE_OPENSSL, ANDROID, IN_SOURCE_BUILD

could this be written to make more sense?

Reply:
once again. Yes. After targets are fixed though?